### DISCUSSION TITLE
Pack Size usage

### DISCUSSION THREAD

**Discussion by temmeand** [inventree/InvenTree#4282] [Q&A] [answered]
Thanks for adding the pack size feature. Question on how it should be used.

Let's start by looking at a 25L tin of Yellow Paint in the the demo instance, [Supplier Part `PNT.YELLOW.25L`](https://demo.inventree.org/supplier-part/1011/). I see 2 stock items for this, [2500L](https://demo.inventree.org/stock/item/1001/) and [125L](https://demo.inventree.org/stock/item/1000/) from PO0005. If I add stock the 2500L, I would expect to be adding the number of packs, not entering the number of liters since I am editing the stock for a 25L tin. Adding a stock of 10 results in 2510L, not 2750L.

Likewise, if I create a new Stock Item, the quantity is liters, not packs. (New Stock Item, Quantity 5 gives 5L, not 125L).

What is the expected behavior for this use case as currently developed? Did I find an edge case or is my use case not covered?

Without digging into the neither code nor the past discussions (I've browsed them, but not closely read), it may be appropriate to change the term used for stock to `Packs` instead of `Quantity` when a Pack Size is specified.

**Comment by miggland** [comment, marked as answer]
The behaviour is as expected when you add via a purchase order - here the terminology is clear, and adds the expected amounts, for example buying 10 packs of 25 L paint gives you 250 L more in stock.

I agree though that the behaviour is unexpected when adding stock directly. Here, what you add is in the units of the base part, in this case liters. I say unexpected because the supplier part has a pack size defined, which is not taken into account.

I would suggest adding a FR to remove the unexpected behaviour: pack sizes should be taken into account when adding stock.

Just on the side: If using the pack quantity like this in for paint makes sense or not is a question of your use case. If you're using paint yourself, you'll probably only be interested in the total amount of paint, in litres. If you're selling paint, you can't necessarily split up 25 L of paint into smaller units, so you might track different parts for each tin size (not just supplier parts).

**Reply by SchrodingersGat** [reply]
Thanks @miggland for the response :) you are on the money!
